src/coprocessor/dag/expr/ctx.rs

```diff
@@ -47,6 +47,113 @@ pub const FLAG_DIVIDED_BY_ZERO_AS_WARNING: u64 = 1 << 8;
 pub const MODE_ERROR_FOR_DIVISION_BY_ZERO: u64 = 27;
 
 const DEFAULT_MAX_WARNING_CNT: usize = 64;
+
+pub struct EvalConfigBuilder {
+    pub tz: FixedOffset,
+    pub ignore_truncate: bool,
+    pub truncate_as_warning: bool,
+    pub overflow_as_warning: bool,
+    pub in_insert_stmt: bool,
+    pub in_update_or_delete_stmt: bool,
+    pub in_select_stmt: bool,
+    pub pad_char_to_full_length: bool,
+    pub divided_by_zero_as_warning: bool,
+    pub max_warning_cnt: usize,
+    pub sql_mode: u64,
```

Comment:
Please also provide a method of directly setting flags, since it is what we use normally.

Reply:
Fixed